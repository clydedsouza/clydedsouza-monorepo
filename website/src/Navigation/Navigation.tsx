import { Outlet, Link } from "react-router-dom";

function Navigation() {
    return (
      <div>
       <nav>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/projects">Blogs</Link>
          </li>
          <li>
            <Link to="/headline">Healdine</Link>
          </li>
        </ul>
      </nav>

      <Outlet />
      </div>
    );
  }
  
  export default Navigation;