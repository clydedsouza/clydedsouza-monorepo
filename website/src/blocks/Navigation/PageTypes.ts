export enum PageTypes {
  About = "About",
  Portfolio = "Portfolio",
  Platforms = "Platforms",
  Highlights = "Highlights",
  Cta = "Cta",
  Blog = "Blog",
}
