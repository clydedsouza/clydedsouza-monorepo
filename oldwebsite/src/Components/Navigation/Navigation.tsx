import { HashRouter, NavLink, Outlet, Route, Routes } from 'react-router-dom'
import { sendNavigationClickedEvent } from '../../Api/Analytics'
import AboutMe from '../../Pages/AboutMe/AboutMe'
import Platforms from '../../Pages/Platforms/Platforms'
import Portfolio from '../../Pages/Portfolio/Portfolio'
import './Navigation.scss'
import { PageTypes } from './PageTypes'

function Menu() {
  return (
    <>
      <nav>
        <NavLink
          to="/"
          className={({ isActive }) => (isActive ? 'active' : '')}
          onClick={() => sendNavigationClickedEvent(PageTypes.About)}
        >
          {PageTypes.About}
        </NavLink>
        <NavLink
          to={PageTypes.Portfolio.toLowerCase()}
          className={({ isActive }) => (isActive ? 'active' : '')}
          onClick={() => sendNavigationClickedEvent(PageTypes.Portfolio)}
        >
          {PageTypes.Portfolio}
        </NavLink>
        <NavLink
          to={PageTypes.Platforms.toLowerCase()}
          className={({ isActive }) => (isActive ? 'active' : '')}
          onClick={() => sendNavigationClickedEvent(PageTypes.Platforms)}
        >
          {PageTypes.Platforms}
        </NavLink>
      </nav>
      <Outlet />
    </>
  )
}

function Navigation() {
  return (
    <HashRouter>
      <Routes>
        <Route path="/" element={<Menu />}>
          <Route index element={<AboutMe />} />
          <Route
            path={PageTypes.Portfolio.toLowerCase()}
            element={<Portfolio />}
          />
          <Route
            path={PageTypes.Platforms.toLowerCase()}
            element={<Platforms />}
          />
          <Route path="*" element={<AboutMe />} />
        </Route>
      </Routes>
    </HashRouter>
  )
}

export default Navigation
