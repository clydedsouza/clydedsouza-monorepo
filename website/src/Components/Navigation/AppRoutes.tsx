import { BrowserRouter, NavLink, Outlet, Route, Routes } from 'react-router-dom'
import AboutMe from '../../Pages/AboutMe/AboutMe'
import Projects from '../../Pages/Projects/Projects'
import { PageTypes } from '../../Types/PageTypes'

function Navigation() {
  return (
    <>
      <nav>
        <NavLink
          to="/"
          className={({ isActive }) => (isActive ? 'active' : '')}
        >
          {PageTypes.About}
        </NavLink>
        <NavLink
          to="/highlights"
          className={({ isActive }) => (isActive ? 'active' : '')}
        >
          {PageTypes.Highlights}
        </NavLink>
        <NavLink
          to="/platforms"
          className={({ isActive }) => (isActive ? 'active' : '')}
        >
          {PageTypes.Platforms}
        </NavLink>
      </nav>
      <Outlet />
    </>
  )
}

function AppRoutes() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Navigation />}>
          <Route index element={<AboutMe />} />
          <Route
            path={PageTypes.Highlights.toLowerCase()}
            element={<Projects name={PageTypes.Highlights} />}
          />
          <Route
            path={PageTypes.Platforms.toLowerCase()}
            element={<Projects name={PageTypes.Platforms} />}
          />
          {/* <Route path="*" element={<NoPage />} /> */}
        </Route>
      </Routes>
    </BrowserRouter>
  )
}

export default AppRoutes
