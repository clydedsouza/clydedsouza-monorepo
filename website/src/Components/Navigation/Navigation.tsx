import { BrowserRouter, NavLink, Outlet, Route, Routes } from 'react-router-dom'
import AboutMe from '../../Pages/AboutMe/AboutMe'
import Projects from '../../Pages/Projects/Projects'
import { PageTypes } from '../../Types/PageTypes'
import './Navigation.scss'

function Menu() {
  return (
    <>
      <nav>
        <NavLink
          to="/"
          className={({ isActive }) => (isActive ? 'active' : '')}
        >
          {PageTypes.About}
        </NavLink>
        <NavLink
          to={PageTypes.Highlights.toLowerCase()}
          className={({ isActive }) => (isActive ? 'active' : '')}
        >
          {PageTypes.Highlights}
        </NavLink>
        <NavLink
          to={PageTypes.Platforms.toLowerCase()}
          className={({ isActive }) => (isActive ? 'active' : '')}
        >
          {PageTypes.Platforms}
        </NavLink>
      </nav>
      <Outlet />
    </>
  )
}

function Navigation() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Menu />}>
          <Route index element={<AboutMe />} />
          <Route
            path={PageTypes.Highlights.toLowerCase()}
            element={<Projects name={PageTypes.Highlights} />}
          />
          <Route
            path={PageTypes.Platforms.toLowerCase()}
            element={<Projects name={PageTypes.Platforms} />}
          />
          <Route path="*" element={<AboutMe />} />
        </Route>
      </Routes>
    </BrowserRouter>
  )
}

export default Navigation
