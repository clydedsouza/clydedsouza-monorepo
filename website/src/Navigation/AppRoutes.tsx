import { Routes, Route, BrowserRouter } from 'react-router-dom'
import AboutMe from '../AboutMe/AboutMe'
import Headline from '../Headline/Headline'
import ProjectTile from '../ProjectTile/ProjectTile'
import Navigation from './Navigation'

function AppRoutes() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Navigation />}>
          <Route index element={<AboutMe />} />
          <Route path="headline" element={<Headline />} />
          <Route
            path="projects"
            element={<ProjectTile name={''} image={''} description={''} />}
          />
          {/* <Route path="*" element={<NoPage />} /> */}
        </Route>
      </Routes>
    </BrowserRouter>
  )
}

export default AppRoutes
