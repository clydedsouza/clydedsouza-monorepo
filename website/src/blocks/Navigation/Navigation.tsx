"use client";

import { sendNavigationClickedEvent } from "@/api/Analytics";
import Link from "next/link";
import { usePathname } from "next/navigation";
import { useState } from "react";
import "./Navigation.scss";
import { PageTypes } from "./PageTypes";

const getCurrentPage = (pathFromNextJs: string) => {
  const cleanPath = pathFromNextJs.replaceAll("/", "").toLowerCase();

  if (cleanPath === PageTypes.Platforms.toLowerCase()) {
    return PageTypes.Platforms;
  } else if (cleanPath === PageTypes.Portfolio.toLowerCase()) {
    return PageTypes.Portfolio;
  } else {
    return PageTypes.About;
  }
};

function Navigation() {
  const [currentPage, setCurrentPage] = useState(getCurrentPage(usePathname()));

  const onNavigationItemClick = (pageType: PageTypes) => {
    sendNavigationClickedEvent(pageType);
    setCurrentPage(pageType);
  };

  return (
    <>
      <nav>
        <Link
          href="/"
          className={currentPage === PageTypes.About ? "active" : ""}
          onClick={() => onNavigationItemClick(PageTypes.About)}
        >
          {PageTypes.About}
        </Link>
        <Link
          href={PageTypes.Portfolio.toLowerCase()}
          className={currentPage === PageTypes.Portfolio ? "active" : ""}
          onClick={() => onNavigationItemClick(PageTypes.Portfolio)}
        >
          {PageTypes.Portfolio}
        </Link>
        <Link
          href={PageTypes.Platforms.toLowerCase()}
          className={currentPage === PageTypes.Platforms ? "active" : ""}
          onClick={() => onNavigationItemClick(PageTypes.Platforms)}
        >
          {PageTypes.Platforms}
        </Link>
        <Link
          href={"https://blog.clydedsouza.net"}
          target="_blank"
          onClick={() => sendNavigationClickedEvent(PageTypes.Blog)}
        >
          {PageTypes.Blog}
        </Link>
      </nav>
    </>
  );
}

export default Navigation;
