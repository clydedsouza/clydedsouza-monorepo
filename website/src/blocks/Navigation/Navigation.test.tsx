import { render } from "@testing-library/react";
import Navigation from "./Navigation";

describe("Navigation", () => {
  it("should render navigation", async () => {
    render(<Navigation />);
    expect(document.body).toMatchSnapshot();
  });
});
