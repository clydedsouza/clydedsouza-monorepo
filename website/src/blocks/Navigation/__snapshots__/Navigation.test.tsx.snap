// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render navigation 1`] = `
<body>
  <div>
    <nav>
      <a
        class="active"
        href="/"
      >
        About
      </a>
      <a
        class=""
        href="portfolio"
      >
        Portfolio
      </a>
      <a
        class=""
        href="platforms"
      >
        Platforms
      </a>
    </nav>
  </div>
</body>
`;
